src/services/recim/activity.ts

```diff
@@ -0,0 +1,83 @@
+import http from '../http';
+import { Sport } from './sport';
+import { Team } from './team';
+import { Series } from './series';
+
+export type Activity = {
+  ID: number;
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  RegistrationOpen: boolean;
+  SportID: Sport;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+  Series: Series[];
+  Team: Team[];
+};
+
+type CreatedActivity = {
+  ID: number;
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: Sport;
+  StatusID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Completed: boolean;
+};
+
+type UploadActivity = {
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+};
+
+type PatchActivity = {
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: number;
+  StatusID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+  Completed: boolean;
+};
+
+//Activity Routes
+const createActivity = (newActivity: UploadActivity): Promise<CreatedActivity> =>
+  http.post('recim/activities', newActivity);
+
+const getActivityByID = (ID: number): Promise<Activity> => http.get(`recim/activities/${ID}`);
+
+const getAllActivities = (
+  active: boolean,
+  time: String,
+  registrationOpen: boolean,
+): Promise<Activity[]> => {
+  if (registrationOpen) {
+    return http.get(`recim/activities?registrationOpen=${registrationOpen}`);
+  } else {
+    if (time) {
+      return http.get(`recim/activities?active=${active}&time=${time}`);
+    } else {
+      return http.get(`recim/activities?active=${active}`);
+    }
+  }
```

Comment:
If you just add all of the potential parameters to the URL, the nulls will just appear as null to the API. You can simplify to:
```suggestion
  return http.get(`recim/activities?active=${active}&time=${time}&registrationOpen=${registrationOpen}`);
```